metaflow/plugins/aws/batch/batch_client.py

```diff
@@ -261,6 +262,23 @@ def _register_job_definition(
                     {"sourceVolume": name, "containerPath": host_path}
                 )
 
+        if secrets:
+            job_definition["containerProperties"]["secrets"] = []
+            if isinstance(secrets, str):
+                secrets = [secrets]
+            for secret in secrets:
+                delimiter = ","
```

Comment:
I'm not sure what best practice for user-supplied key-value pairs is.

I primarily worked with this example when testing:

```
python flow.py run --with batch:secrets=secret_a,arn:aws:secretsmanager:secret_a
```

I originally tried `=` and `->` but both proved problematic with how Metaflow seems to be handling incoming arguments.

Reply:
@stevenhoelscher Would it make sense to do `python flow.py run --with batch:secrets=arn:aws:secretsmanager:secret_a` which will expose `secret_a`? That way we don't have to maintain a mapping. Wdyt?

Reply:
Happy to chat live in the `#dev-metaflow` channel in chat.metaflow.org.

Reply:
Based on [Slack conversation](https://outerbounds-community.slack.com/archives/C020U025QJK/p1647038067402739), updated secrets type to `JSONType` / dictionary